base/printf.jl

```diff
@@ -16,14 +16,15 @@ function gen(s::AbstractString)
             push!(blk.args, :(write(out, $(length(x)==1 ? x[1] : x))))
         else
             c = lowercase(x[end])
-            f = c=='f' ? gen_f :
-                c=='e' ? gen_e :
-                c=='a' ? gen_a :
-                c=='g' ? gen_g :
-                c=='c' ? gen_c :
-                c=='s' ? gen_s :
-                c=='p' ? gen_p :
-                         gen_d
+            if c=='f'; f = gen_f
+            elseif c=='e'; f = gen_e
+            elseif c=='a'; f = gen_a
+            elseif c=='g'; f = gen_g
+            elseif c=='c'; f = gen_c
+            elseif c=='s'; f = gen_s
+            elseif c=='p'; f = gen_p
+            else f = gen_d end
+            
```

Comment:
This line has trailing whitespace. Some programmers consider that bad, because it often litters the diffs with unrelated changes the next time someone edits this file.

As a result we added a check in the CI system so that tests fails when trailing whitespace is detected, so this needs to be fixed before the tests can execute successfully. @hayd has a great suggestion for how to let git automatically fix any such issues.